engines/hypno/scene.cpp

```diff
@@ -0,0 +1,372 @@
+
+	stack.clear();
+	_nextHotsToAdd = &scene.hots;
+	defaultCursor();
+
+	while (!shouldQuit() && _nextLevel.empty()) {
+
+		while (g_system->getEventManager()->pollEvent(event)) {
+			mousePos = g_system->getEventManager()->getMousePos();
+			// Events
+			switch (event.type) {
+			case Common::EVENT_KEYDOWN:
+				if (event.kbd.keycode == Common::KEYCODE_ESCAPE) {
+					for (Videos::iterator it = _videosPlaying.begin(); it != _videosPlaying.end(); ++it) {
+						if (it->decoder)
+							skipVideo(*it);
+					}
+					_videosPlaying.clear();
+
+					if (!stack.empty()) {
+						runMenu(*stack.back());
+						drawScreen();
+					}
+				}
+
+				break;
+
+			case Common::EVENT_QUIT:
+			case Common::EVENT_RETURN_TO_LAUNCHER:
+				break;
+
+			case Common::EVENT_RBUTTONDOWN:
+				if (stack.empty())
+					break;
+				if (!_conversation.empty()) {
+					rightClickedConversation(mousePos);
+					break;
+				}
+				break;
+
+			case Common::EVENT_LBUTTONDOWN:
+				if (stack.empty())
+					break;
+				if (!_conversation.empty()) {
+					leftClickedConversation(mousePos);
+					break;
+				}
+				if (!_nextHotsToAdd && !_nextHotsToRemove /*&& _videosPlaying.size() == 0*/)
+					clickedHotspot(mousePos);
+				break;
+
+			case Common::EVENT_MOUSEMOVE:
+				// Reset cursor to default
+				//changeCursor("default");
+				// The following functions will return true
+				// if the cursor is changed
+				if (stack.empty() || !_conversation.empty() || _videosPlaying.size() > 0)
+					break;
+
+				if (hoverHotspot(mousePos)) {
+				} else
+					defaultCursor();
+				break;
+
+			default:
+				break;
+			}
+		}
+
+		if (_refreshConversation && !_conversation.empty() && _nextSequentialVideoToPlay.empty()) {
+			showConversation();
+			drawScreen();
+			_refreshConversation = false;
+			_videosPlaying.clear();
+		}
+
+		// Movies
+		if (_nextSequentialVideoToPlay.size() > 0 && _videosPlaying.empty()) {
+			playVideo(*_nextSequentialVideoToPlay.begin());
+			_videosPlaying.push_back(*_nextSequentialVideoToPlay.begin());
+			_nextSequentialVideoToPlay.remove_at(0);
+		}
+		uint32 i = 0;
+		videosToRemove.clear();
+		for (Videos::iterator it = _videosPlaying.begin(); it != _videosPlaying.end(); ++it) {
+
+			if (it->decoder) {
+				if (it->decoder->endOfVideo()) {
+					if (it->loop) {
+						it->decoder->rewind();
+						it->decoder->start();
+					} else {
+						it->decoder->close();
+						delete it->decoder;
+						it->decoder = nullptr;
+						videosToRemove.push_back(i);
+					}
+
+				} else if (it->decoder->needsUpdate()) {
+					updateScreen(*it);
+				}
+			}
+			i++;
+		}
+		if (videosToRemove.size() > 0) {
+
+			for (Common::List<uint32>::iterator it = videosToRemove.begin(); it != videosToRemove.end(); ++it) {
+				debugC(1, kHypnoDebugScene, "removing %d from %d size", *it, _videosPlaying.size());
+				_videosPlaying.remove_at(*it);
+			}
+
+			// Nothing else to play
+			if (_videosPlaying.size() == 0 && _nextSequentialVideoToPlay.size() == 0) {
```

Comment:
There is a method for performing the empty check (`empty()`). This has the benefit that is you are ever going to change the container type of the videos the call is still `O(1)` and you don't have to compute the size (this is of course just theoretical because `Common::Array` doesn't compute its size with each call, but the style issue still counts)

Reply:
Fixed.